primitives/core/src/ed25519.rs

```diff
@@ -484,10 +486,11 @@ impl TraitPair for Pair {
 	///
 	/// You should never need to use this; generate(), generate_with_phrase
 	fn from_seed_slice(seed_slice: &[u8]) -> Result<Pair, SecretStringError> {
-		let secret = ed25519_dalek::SecretKey::from_bytes(seed_slice)
+		// does try_into consume the seed? can I consume seed_slice here? I think not right?
```

Comment:
You can't consume the seed in this fn but have raised a PR to make from_seed (above) take ownership of the Seed.
https://github.com/paritytech/substrate/pull/9683

That then allows `SigningKey::from(seed)` in the from_seed function.

Reply:
I agree with your point in that PR that the seed should be consumed; will update this if that gets merged